### ISSUE TITLE
Tags, custom fields and version control

### ISSUE THREAD

**Issue by pulsarinformatique** [joomla/joomla-cms#15700]
Hi
Seems that any change on the tags or custom fields value in an article doesn't generate a new version of the article.

Steps to reproduce the issue

Create a custom field (any type) and then create an article with this custom field. save it. now change the value of the custom field and check if a new version of the article has been generated (with the VERSION button).

Expected result

When you change the tags or the values of a custom field inside an article we may expect a new version to be generated

Actual result

no new version is generated. the control version only applies to the #__content table

thanks

cyril

System information (as much as possible)

Additional comments

**Comment by PhilETaylor** [comment]
I raised this a long time ago - and I was told to shut up, this is how it was designed, and this is how it will stay because the developers of those features totally disregarded the versioning system in their architecture and therefore no one can be bothered to re-architecture the features so that they can be versioned...

**Comment by PhilETaylor** [comment]
https://github.com/joomla/joomla-cms/issues/13863

**Comment by Bakual** [comment]
I think tags and versioning will be re-architectured for J4. Maybe it's possible then. For J3 and the way those features work it's just not possible to do as far as I can imagine. If you have an idea, feel free to share 👍

**Comment by pulsarinformatique** [comment]
It's a pity we have to wait for J4 for a more integrated solution.

Tags and custom fields are now completely ignored for the version control.
The same thing could be said for the lists parameters and the custom fields.

I feel we have much too separated great tools (articles, tags, fields) at our disposal.

thanks

cyril

**Comment by Bakual** [comment]
> Tags and custom fields are now completely ignored for the version control.

> I feel we have much too separated great tools (articles, tags, fields) at our disposal.

Actually I think fields is integrated fine, the issue is that the version feature works from the table class and thus can only version control the properties of that class. Since fields a re processed outside of that table class (which is fine), it doesn't get version controlled. The same issue actually exists for extensions which store an item form into multiple tables. It's a limitation of the current version feature.

**Comment by pulsarinformatique** [comment]
I agree with you. From a webmaster point of view, whether it comes from the control version or the tag implementation, we feel we are lacking something.

when you change a tag or a custom field in an article you expect it generates a new version

**Comment by PhilETaylor** [comment]
> I think tags and versioning will be re-architectured for J4.

No evidence for that statement. 

Too many people suggesting that Joomla 4 will solve everyones hopes and dreams.

**Comment by Bakual** [comment]
> No evidence for that statement.

Yep, no evidence, but I saw two new plugins "taggable" and "versionable" in a new plugin group "behaviour" and the observer classes are gone. This made me believe something changed (and then "hope" started).

**Comment by mbabker** [comment]
What changed is the observable pattern in JTable being replaced with "regular" plugin events.  IIRC pretty much everything that didn't have the "normal" event dispatching setup (i.e. JEditor) has been updated to use our (now) singular event dispatcher and plugins.

**Comment by pulsarinformatique** [comment]
Hi
Do we have a clearer view now of what could be expected in J4 regarding the tags / custom fields modifications and version control ?

thanks

**Comment by Bakual** [comment]
Not that I'm aware of. And as long as nobody takes up the work to rewrite the version history, nothing will change.

**Comment by brianteeman** [comment]
please change label to J4

@HLeithner @bembelimen @nibra Reading the comments above it sounds like it would need considerable work to achieve this (good idea) and perhaps this is something for j5 OR something you explicitly state will not happen at all.

**Comment by Hackwar** [comment]
Versioning for tags and custom fields will come eventually. Yet again something someone has to dive into, but what also isn't trivial to do. Changing label from J3 to J4.
